expression/integration_test.go

```diff
@@ -7113,3 +7113,19 @@ func (s *testIntegrationSerialSuite) TestIssue17063(c *C) {
 	tk.MustQuery(`select collation(lag(b, 1, 'B') over w) from t window w as (order by b);`).Check(testkit.Rows("utf8mb4_general_ci", "utf8mb4_general_ci"))
 	tk.MustQuery(`select coercibility(lag(b, 1, 'B') over w) from t window w as (order by b);`).Check(testkit.Rows("2", "2"))
 }
+
+func (s *testIntegrationSerialSuite) TestIssue18949(c *C) {
+	collate.SetNewCollationEnabledForTest(true)
+	defer collate.SetNewCollationEnabledForTest(false)
+
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`use test;`)
+	tk.MustExec(`drop table if exists t;`)
+	tk.MustExec(`create table t(a enum('1 ', '1\t', ' 1 '), b set('1 ', '1\t', ' 1 '));`)
+	tk.MustQuery("show create table t").Check(
```

Comment:
We have better use `c.Assert(result, Matches, "regex_pattern")` style. If we change the other part in `show create table`, we don't need to modify this test again.

Reply:
updated